src/ocaml-index/lib/index.ml

```diff
@@ -40,10 +40,14 @@ let gather_locs_from_fragments ~root ~rewrite_root map fragments =
   in
   Shape.Uid.Tbl.fold add_loc fragments map
 
-module Reduce_conf = struct
+module Reduce_conf (Loaded_shapes : sig
+  val shapes : (string, Shape.t) Hashtbl.t
+end) = struct
   let fuel = 10
 
   let try_load ~unit_name () =
+    match Hashtbl.find_opt Loaded_shapes.shapes unit_name with
+    | None -> begin
     let artifact =
```

Comment:
I recognize the indentation here is sad. I do this so that the diff from upstream is smaller. If I upstream this change, I'll fix the indentation.

Reply:
done